tk_toolchain/tk_testengine/startup.py

```diff
@@ -0,0 +1,70 @@
+# Copyright (c) 2017 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import os
+import json
+
+from sgtk.platform import SoftwareLauncher, SoftwareVersion, LaunchInformation
+
+
+class TestLauncher(SoftwareLauncher):
+    def scan_software(self):
+        """
+        For each software executable that was found, get the software products for it.
+
+        :returns: List of :class:`SoftwareVersion`.
+        """
+
+        # A list of software scan results can be provided via the SHOTGUN_SCAN_SOFTWARE_LIST env var
+        # as a serialized json string in the following format
+        # [
+        #     [
+        #         "2050", # version
+        #         "Software Name",
+        #         "path/to/executable",
+        #         "path/to/icon", # optional can be empty string
+        #         ["arg1", "arg2"], # args, can be an empty list
+        #     ]
+        # ]
+
+        software_list = []
+
+        serialized_software_list = os.environ.get("SHOTGUN_SCAN_SOFTWARE_LIST")
+        imported_software_list = json.loads(serialized_software_list)
+        if imported_software_list:
+            for scanned_software in imported_software_list:
+                sw = SoftwareVersion(*scanned_software)
```

Comment:
Would the interface be even simpler here if we simply pickled `SoftwareVersion` objects from the test code?

Reply:
yeah perhaps, I'll look at doing that

Reply:
Done, yeah I think that is better